examples/games/game_menu.rs

```diff
@@ -397,380 +444,157 @@ mod menu {
-                                    button_text_style.clone(),
-                                ));
-                            });
-                        parent
-                            .spawn((
-                                ButtonBundle {
-                                    style: button_style.clone(),
-                                    background_color: NORMAL_BUTTON.into(),
-                                    ..default()
-                                },
-                                MenuButtonAction::Settings,
-                            ))
-                            .with_children(|parent| {
-                                let icon = asset_server.load("textures/Game Icons/wrench.png");
-                                parent.spawn(ImageBundle {
-                                    style: button_icon_style.clone(),
-                                    image: UiImage::new(icon),
-                                    ..default()
-                                });
-                                parent.spawn(TextBundle::from_section(
-                                    "Settings",
-                                    button_text_style.clone(),
-                                ));
-                            });
-                        parent
-                            .spawn((
-                                ButtonBundle {
-                                    style: button_style,
-                                    background_color: NORMAL_BUTTON.into(),
-                                    ..default()
-                                },
-                                MenuButtonAction::Quit,
-                            ))
-                            .with_children(|parent| {
-                                let icon = asset_server.load("textures/Game Icons/exitRight.png");
-                                parent.spawn(ImageBundle {
-                                    style: button_icon_style,
-                                    image: UiImage::new(icon),
-                                    ..default()
-                                });
-                                parent.spawn(TextBundle::from_section("Quit", button_text_style));
-                            });
                     });
-            });
+                    parent.spawn(TextBundle::from_section(text, button_text_style()));
+                });
+            }
+        });
     }
 
     fn settings_menu_setup(mut commands: Commands) {
-        let button_style = Style {
-            width: Val::Px(200.0),
-            height: Val::Px(65.0),
-            margin: UiRect::all(Val::Px(20.0)),
-            justify_content: JustifyContent::Center,
-            align_items: AlignItems::Center,
-            ..default()
-        };
-
-        let button_text_style = TextStyle {
-            font_size: 40.0,
-            color: TEXT_COLOR,
-            ..default()
-        };
+        let red = Color::CRIMSON;
```

Comment:
ahah, I did this because otherwise the function call directly following gets expanded on multiple lines, which increases the indentation by two, because of the final argument being a closure.

Not sure how to improve that.

Maybe have all the colors moved into top level `const`s? It's pretty good practice to create your own "palette" in a single location and re-use it for styling.

Reply:
(personally I'd just let `rustfmt` do its thing, and if the resulting formatting is ugly it's probably a sign that the code itself is ugly, and maybe it would be better for these helper functions to return an `Entity` that you can save to a named variable and then do something like:

```rust
let button = spawn_button(...);
commands.entity(button).set_parent(settings_menu);
```

inside the for-loop instead of passing around closures and `ChildBuilder`... but that's a much bigger suggestion than I'd like to make)

Top-level `const`s sounds reasonable, since this color is used in multiple places.

Reply:
created top-level `const RED: Color = Color::CRIMSON;`